### PR TITLE
updtae trigger

### PR DESCRIPTION
**Pull request by drillinP**
Update du trigger pour calcul de point apres validation

### CODE REVIEW

**Comment by mdouchin** [comment]
@Gustry I tested locally this branch, and `make tests` runs OK. But Travis complains :(  Cannot figure out why

**Comment by Gustry** [comment]
It's not `make test` which is failing. `make test` is doing Python based tests.

What is failing here is `make test_migration`, the migration test.

How to fix the PR and make Travis happy:
* to have a clean git status
* `make test_migration`
* and commit again

**Comment by drillinP** [comment]
@Gustry 
* I have a clean git status
* make test_migration shows an error again

**Comment by rldhont** [comment]
@pdrillin peux-tu ajouter un test ?

File: gestion_base_adresse/install/sql/adresse/10_FUNCTION.sql [hunk 1]
```diff
@@ -208,11 +208,23 @@ DECLARE
...
+      IF NEW.id_voie IS NULL THEN
         SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
         UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
-    ELSIF TG_OP = 'INSERT' THEN
+      ELSE
         SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
         UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
+        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
+        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
+      END IF;
```
Comment (rldhont) [inline gestion_base_adresse/install/sql/adresse/10_FUNCTION.sql:227]:
```suggestion
      ELSIF OLD.id_voie IS NULL THEN
          SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
          UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
      ELSE
        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
      END IF;
```

File: gestion_base_adresse/install/sql/upgrade/upgrade_to_0.5.0.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ELSIF TG_OP = 'UPDATE' AND NEW.id_voie IS DISTINCT FROM OLD.id_voie THEN
+      IF NEW.id_voie IS NULL THEN
+        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
+        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
+      ELSE
+        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
+        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
+        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
+        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
+      END IF;
```
Comment (rldhont) [inline gestion_base_adresse/install/sql/upgrade/upgrade_to_0.5.0.sql:26]:
```suggestion
      ELSIF OLD.id_voie IS NULL THEN
          SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
          UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
      ELSE
        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = NEW.id_voie;
        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = NEW.id_voie;
        SELECT COUNT(id_point) into nb FROM adresse.point_adresse WHERE id_voie = OLD.id_voie;
        UPDATE adresse.voie SET nb_point = nb WHERE id_voie = OLD.id_voie;
      END IF;
```

**Comment by rldhont** [comment]
@drillinP lgtm

**Comment by drillinP** [comment]
@rldhont ca veut dire quoi lgtm ?

**Pull request merged**